deps/spidershim/src/v8context.cc

```diff
@@ -139,6 +138,52 @@ void Context::Exit() {
   GetIsolate()->PopCurrentContext();
 }
 
+void Context::SetEmbedderData(int idx, Local<Value> data)
+{
+	assert(idx >= 0);
+  if (pimpl_->embedderData.length() <= static_cast<unsigned int>(idx) &&
+      !pimpl_->embedderData.resize(idx + 1)) {
+    return;
+  }
+
+  pimpl_->embedderData[idx].set(*GetValue(data));
+}
+
+Local<Value>
+Context::GetEmbedderData(int idx)
+{
+	assert(idx >= 0);
+  if (static_cast<unsigned int>(idx) >= pimpl_->embedderData.length()) {
+    return Local<Value>();
+  }
+
+  return Local<Value>::New(Isolate::GetCurrent(), GetV8Value(pimpl_->embedderData[idx]));
+}
+
+void
+Context::SetAlignedPointerInEmbedderData(int idx, void* data)
+{
+  assert((reinterpret_cast<uintptr_t>(data) & 0x1) == 0);
```

Comment:
V8 does something to the effect of MOZ_RELEASE_ASSERT here.

Also if I understand their setup correctly I think they enforce 4-byte aligned...  But chakrashim seems to get away without any checks whatsoever!